### ISSUE TITLE
Yarn missing module files generated from 'prepare' hook

### ISSUE THREAD

**Issue by kicktheken** [yarnpkg/yarn#7778]
Bug: `yarn` missing module files generated from 'prepare' hook

## Setup
Yarn v1.21.1
macOS 10.15.2
node v12.13.1

## Steps to Reproduce 

`yarn add https://github.com/kicktheken/PIXI.TextInput`

## Expected behavior

```
> find node_modules/pixi-text-input/dist
node_modules/pixi-text-input/dist
node_modules/pixi-text-input/dist/app
node_modules/pixi-text-input/dist/app/TextInput.js
node_modules/pixi-text-input/dist/app/DefaultBoxGenerator.d.ts
node_modules/pixi-text-input/dist/app/DefaultBoxGenerator.js
node_modules/pixi-text-input/dist/app/TextInput.d.ts
node_modules/pixi-text-input/dist/index.js
node_modules/pixi-text-input/dist/index.d.ts
```
## Actual behavior

```
> find node_modules/pixi-text-input/dist
node_modules/pixi-text-input/dist
node_modules/pixi-text-input/dist/index.js
```
I did `npm i https://github.com/kicktheken/PIXI.TextInput` (with npm v6.13.4) and the expected files exists so this looks like a bug in `yarn`

**Comment by rally25rs** [comment]
duplicate of #5235 which looks like it would be addressed by #6131 but it has been sitting open for quite a while.

**Comment by tpai** [comment]
Just in case someone came for the solution, it seems Yarn respect `.gitignore` while building artifact, that cause everything inside `dist/` will be ignored except the file which specified at **main** field of `package.json` which is `dist/index.js`.

According to #5235 , there are three workaround for this:
- Modify `.gitignore` to not ignore `dist/`.
- Add `.npmignore`, Yarn respect it over `.gitignore`.
- Specify `dist/` at **files** field in `package.json`.
